packages/squiggle-lang/src/rescript/ReducerInterface/ReducerInterface_FunctionParser.res

```diff
@@ -0,0 +1,103 @@
+type expressionValue = ReducerInterface_ExpressionValue.expressionValue
+type error = DistributionTypes.error
+type argumentError = DistributionTypes.ArgumentError.t
+
+let expressionValueToType = (value: expressionValue): DistributionTypes.expressionType =>
+  switch value {
+  | EvArray(_) => ArrayType
+  | EvArrayString(_) => ArrayStringType
+  | EvBool(_) => BoolType
+  | EvCall(_) => CallType
+  | EvDistribution(_) => DistributionType
+  | EvLambda(_) => LambdaType
+  | EvNumber(_) => NumberType
+  | EvRecord(_) => RecordType
+  | EvString(_) => StringType
+  | EvSymbol(_) => SymbolType
+  }
+
+module Primitive = {
+  let distribution = (argument: expressionValue): result<DistributionTypes.genericDist, argumentError> =>
+    switch argument {
+    | EvDistribution(dist) => Ok(dist)
+    | _ => Error(WrongTypeError(DistributionType, expressionValueToType(argument)))
+    }
+
+  let finite = (argument: expressionValue): result<SafeFloat.finite, argumentError> =>
+    switch argument {
+    | EvNumber(num) => 
+        switch SafeFloat.Finite.make(num) {
+        | Some(safeNum) => Ok(safeNum)
+        | None => Error(MustBeFinite)
```

Comment:
I feel kind of weird that this isn't done in the Finite.make() fn or a similar fn.